chore: add 0.66.x release summary notes (#449)
vegaprotocol/documentation: docs/releases/overview.md

### Pre-release Versions 0.66.0 and 0.66.1 combined | 2023-01-06
This version was released to the Vega testnet on 6 January 2023.

Happy New Year from the Vega team.

> The first testnet deployment of 2023 brings with it the final planned data node stability improvements ahead of the market simulations. Using the data from the last incentive the team has made changes to the way the snapshot metadata is stored, this has significantly improved RAM usage during snapshot file creation. In addition to the memory usage optimisations a panic observed when fetching data segment history has been resolved. 

Reviewer: ```suggestion
The first testnet deployment of 2023 brings with it the final planned data node stability improvements ahead of the market simulations. Using the data from the last incentive, the team has made changes to the way the snapshot metadata is stored, which has significantly improved RAM usage during snapshot file creation. In addition to the memory usage optimisations, a panic when fetching data segment history has been resolved. 
```

These fixes and stability improvements bring the data node software into a state of readiness for the Alpha Mainnet market simulations.

> To round off this release several API improvements have been introduced, along with a number of bug fixes, preparing the core and wallet software for the market simulations. Until then, the team is focused on completing the wallet token support, increasing the test coverage and fixing any bugs found along the way.

Reviewer: ```suggestion
To round off this release several API improvements have been introduced, along with a number of bug fixes, preparing the core and wallet software for the market simulations. Until then, the team is focused on completing wallet API token support, increasing the test coverage and fixing any bugs found along the way.
```

:::caution Breaking changes
**Wallet API  and command removals**: Over the recent testnet releases a number of the wallet APIs have been deprecated. The deployment of `0.66.0` now removes a number of these deprecated fields, endpoints and commands.
**Network parameter name change**: The network parameter `stakeToCcySiskas` has been renamed to `stakeToCcyVolume`.

> **Network parameter name change**: When proposing a market, using a float value for the `triggeringRatio` field, the proposal passes however the response from the APIs was incorrect. This has been resolved by changing the type from `double` to a `string`.

Reviewer: ```suggestion
**Network parameter name change**: The `triggeringRatio` field has been changed from `double` to a `string`. When proposing a market, using a float value would lead to a passing proposal, however the response from the APIs was incorrect. This has been resolved by changing the accepted data format.
```
